pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -125,6 +148,14 @@ var (
 			"upstream_transport_failure_reason": {Kind: &google_protobuf.Value_StringValue{StringValue: "%UPSTREAM_TRANSPORT_FAILURE_REASON%"}},
 		},
 	}
+
+	wellKnownPorts = map[int]protocol.Instance{
+		25:    protocol.TCP,   // SMTP
+		3306:  protocol.MySQL, // MySQL
```

Comment:
is adding all these magic ports a good idea?

Reply:
and if we are doing these why not `80` and `443` which are much more wellKnown? 

Right now istio already does some magic around port 80 and I have heard many people complain about it.

Reply:
IMO, it's a trade-off between allowing protocol sniffing and respecting server-first protocols. If we want to respect server-first protocols, we should not allow protocol sniffing for all well known ports, including 80 and 443. It might restrict the usage of protocol sniffing. If we use protocol sniffing every where, the server-first protocols must be broken. 

In the ideal world, users won't bind service with different protocol to same port, e.g., TCP and HTTP to 3306.

Reply:
I think we need to check the well known ports. Per 8086, @rshriram, WDYT?

Reply:
> I think we need to check the well known ports.

I mean we need to add  to the map the ports which are really needed to be skipped, e.g. 3306.

Reply:
Docker makes it very easy to bind to non-traditional ports (it's all virtualized), so the assumption of [IANA ports](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml) is not as strong as say on VMs.

Reply:
We have numerous complaints about MySQL and Mongo being broken by tls sniffing. Atleast you can disable sniffing for tls using authn apis. Http sniffing has no clean way to disable except killing and restarting pilot with an obscure env var. so we need to take the conservative approach. 

Also keep in mind that this pr is not “enforcing” MySQL filter on 3306. It is simply not setting up protocol sniffing on 3306 irrespective of the type of listener. So the behavior is same as before sniffing was introduced.

Reply:
I agree that we need to be able to opt-out from the inspector on a service/port basis. I raised that point as my first comment, we want protection against mis-detection, whether it's through config or annotation :) I don't agree with these magic numbers, they just add another layer of complexity over the inspector.

Reply:
I changed the code to check port protocol instead of port number.